src/components/FeatureThresholdPanel.tsx

```diff
@@ -85,31 +141,46 @@ export default function FeatureThresholdPanel(inputProps: FeatureThresholdPanelP
   const onClickedRemove = (index: number): void => {
     const newThresholds = [...props.featureThresholds];
     newThresholds.splice(index, 1);
+    // Delete saved min/max bounds for this feature when it's removed.
+    featureMinMax.current.delete(props.featureThresholds[index].featureName);
     props.onChange(newThresholds);
   };
 
+  ////// RENDERING ///////////////////
+  // TODO: Possible bug where selected features with the same name but different units/scaling across
+  // datasets will use the same filter values.
+  // Might not be a problem if collections have similarly-structured datasets...?
+  // The alternative is to use the feature name + units as the key, but that may cause unexpected behavior
+  // for users (such as if one dataset does not have units for a feature but another does)
```

Comment:
suggestion: if feature name matches but units don't, then you just can't keep it

Reply:
Okay, this has been done! It was a bit more involved than I thought it would be. I've updated the verification steps with instructions to test this.